tests/python/pants_test/engine/test_rules.py

```diff
@@ -872,6 +870,7 @@ def g() -> A:
 
         exc_msg = str(cm.exception)
         exc_msg_trimmed = re.sub(r"^.*?(test_rules\.py)", r"\1", exc_msg, flags=re.MULTILINE)
+        self.maxDiff = None
```

Comment:
Left this behind because I found it useful to see why the test was actually failing, even though it wasn't part of the fix per-se